cubi-tools/prototypes/update_metadata.py

```diff
@@ -177,190 +173,88 @@ def clone(workflow_dir, project_dir, ref_repo, source, metadata_dir, dryrun):
-            # an error message that contains the string 'fatal:' will be thrown
-            warning = "fatal:"
-            assert warning not in str(checkout_cmd.stderr.strip()), (
-                "The folder 'template-metadata-files' is not a git repository! "
-                "For this script to work either delete the folder or move it!!"
-            )
-            # If you try to clone a repo/branch/tag that doesn't exist
-            # Git will throw an error message that contains the string 'error:'
-            error = "error:"
-            assert error not in str(
-                checkout_cmd.stderr.strip()
-            ), f"The branch or version tag named '{source}' doesn't exist"
+        if metadata_branch.is_dir():
+            git_pull_template(metadata_branch, branch)
         else:
-            command = [
-                "git",
-                "clone",
-                "-q",
-                "-c advice.detachedHead=false",
-                ref_repo,
-                metadata_dir,
-            ]
-            clone_cmd = sp.run(
-                command,
-                stdout=sp.PIPE,
-                stderr=sp.PIPE,
-                cwd=workflow_dir,
-                check=False,
-            )
-            # If the 'template-metadata-files' folder is not a Git repo
-            # an error message that contains the string 'fatal:' will be thrown
-            warning = "fatal:"
-            assert warning not in str(clone_cmd.stderr.strip()), (
-                "The repository you entered or the branch or version tag "
-                f"named '{source}' doesn't exist"
-            )
-            command_checkout = ["git", "checkout", "".join({source}), "-q"]
-            sp.run(
-                command_checkout,
-                cwd=metadata_dir,
-                check=False,
-            )
+            git_clone_template(ref_repo, metadata_branch, metadata_target, branch)
     return None
 
 
-def get_local_checksum(workflow_dir, f):
-    """
-    TODO: should be refactored, is quasi-identical to 'get_ref_checksum'
-    TODO: single-letter variable name is strongly discouraged
-
-    The MD5 checksum for all metadata files in the
-    local project directory is determined.
+def calculate_md5_checksum(file_path):
     """
-    if workflow_dir.joinpath(f).is_file():
-        with open(workflow_dir.joinpath(f), "rb") as local_file:
-            # read contents of the file
-            local_data = local_file.read()
-            # pipe contents of the file through
-            md5_local = hashlib.md5(local_data).hexdigest()
-    else:
-        md5_local = ""
-    return md5_local
-
+    The MD5 checksum for the metadata files of the local folder or
+    for the template-metadata branch or version tag is determined.
 
-def get_ref_checksum(metadata_dir, f):
-    """
-    TODO: see TODOs in 'get_local_checksum'
+    Args:
+        file_path (pathlib.Path): either the path to metadata_target or metadata_branch
```

Comment:
the argument name suggests that this is always a file, but the documentation says it can also be `metadata_branch` (I assume that to be a directory?).

Reply:
Actually, this is always a file. I clarified the documentation that this is the path to the file either in metadata_target or metadata_branch.